ark/utils/spatial_analysis_utils.py

```diff
@@ -380,3 +382,40 @@ def compute_neighbor_counts(current_fov_neighborhood_data, dist_matrix, distlim,
     freqs = counts.T / num_neighbors
 
     return counts, freqs.T
+
+
+def compute_neighbor_mat_cluster_info(current_neighbor_mat_data, max_k=10, metric='silhouette'):
+    """For a given fov's data in the neighborhood matrix, compute cluster metric scores.
+    Uses k-means clustering.
+
+    Args:
+        current_neighbor_mat_data (pandas.DataFrame):
+            neighborhood matrix data for the current fov
+        max_k (int):
+            the maximum k we want to generate cluster statistics for, must be at least 2
+        metric (str):
```

Comment:
There's a separate 'metric' argument passed directly to silhouette score, which will probably lead to confusion. Lets remove this for now, and once we start supporting multiple different methods, we can add this back in with a new name